schema_utils.py

```diff
@@ -387,3 +411,89 @@ def is_valid_email(obj):
             bool. Whether the given object is a valid email.
         """
         return bool(re.search(r'^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$', obj))
+
+    @staticmethod
```

Comment:
I feel like you need to do a bit more parsing. Doesn't this allow stuff like (352+)-3x which seems misformatted to me?

Probably you'll need to actually build the parse tree corresponding to the expression.

Reply:
I've taken this approach now:
Check if the expression contains invalid characters.
Check for balanced parens. 
Convert the given expression to postfix.
Evaluate the postfix expression and check if a single value is obtained in the end. If so, the expression is valid, else it's not.